[kailashgautham] iP

kailashgautham opened the pull request:
# KaiYap 🚀 
> "_This is the best chatbot I've used, ever!"_ - Kailash Gautham (me) ([source](https://www.youtube.com/watch?v=dQw4w9WgXcQ))

KaiYap is your one-stop solution for all things task-related. Here are some of its features:

1. Add events, todos, and deadlines to your task list.
2. Error handling for those days when you're sleepy and type in the wrong input.
3. Friendly UwU-fied messages for a more personal touch.

"Well, I'm sold! How do I use this?"

- Download our latest release from [this link](https://github.com/kailashgautham/ip/releases).
- Double-click on the application.
- Experience your life re-imagined with task planning at its finest. 

Here are some other **_exciting features_** we're working on!

- [ ] AI-powered task planning
- [ ] A graphical user interface 
- [ ] Calendar features!

```java
class KaiYap {
    public static void main(String[] args) {
        System.out.println("This amazing chatbot is powered by Java!");
    }
}
```

"That's awesome Kailash, I love this app so much that I want to contribute as well!"

Well, fret not! To contribute to this project, please fork the repository, make your changes, and then submit a pull request with a clear description of your improvements. Use this command to fork the repository: 
`git clone https://github.com/kailashgautham/ip.git`

This chatbot is proudly powered by Kailash Gautham & Co.™, dedicated to delivering innovative and intelligent conversational solutions.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

kailashgautham committed:
worked on level-0 tasks

kailashgautham committed:
added echo functionality to chatbot and improved text formatting

kailashgautham committed:
refactored code with functions and added list functionality

kailashgautham committed:
completed marking and unmarking tasks, as well as creating a task class for modularity

kailashgautham committed:
created subclasses todo, event and deadline. improved code structure and implemented inheritance with task class.

kailashgautham committed:
engineered test case for automated testing with bash script

kailashgautham committed:
added error handling and exception classes

kailashgautham committed:
added exception handling and error messages

kailashgautham committed:
added delete functionality

kailashgautham committed:
added task type enum

kailashgautham committed:
completed saving and loading of tasks from file

kailashgautham committed:
completed level 7 tasks - Merge branch 'branch-level-7'

kailashgautham committed:
completed formatting of date and time

kailashgautham committed:
complete level 8 changes - Merge branch 'branch-level-8'

kailashgautham committed:
add classes Ui, Storage, Parser, TaskList

kailashgautham committed:
merged branch A-MoreOOP Merge branch 'A-MoreOOP'

kailashgautham committed:
add package kaiyap

kailashgautham committed:
completed package refactor for A-Packages Merge branch 'A-MoreOOP'

kailashgautham committed:
merge gradle into master - Merge branch 'add-gradle-support'

kailashgautham committed:
add gradle build capability

kailashgautham committed:
added junit tests

kailashgautham committed:
created jar file

kailashgautham committed:
Make styling changes

kailashgautham committed:
Merge branch 'branch-A-CodingStandard' - make coding style changes

kailashgautham committed:
Add JavaDoc for functions, fix coding standard, and add find functionality

kailashgautham committed:
Add checkstyle files and modify var names

kailashgautham committed:
Add checkstyle files - Merge branch 'branch-A-Checkstyle'

alfaloo commented on build.gradle:
```diff
@@ -0,0 +1,47 @@
+
+run{
+    standardInput = System.in
+}
```
I like this code

kailashgautham replied:
Thanks! I appreciate the vote of confidence.

alfaloo commented on gradlew:
```diff
@@ -0,0 +1,248 @@
+        tr '\n' ' '
+    )" '"$@"'
+
+exec "$JAVACMD" "$@"
```
can this be improved?

kailashgautham replied:
I think it's okay! Thanks for taking a look.

alfaloo commented on gradlew.bat:
```diff
@@ -0,0 +1,92 @@
+:mainEnd
+if "%OS%"=="Windows_NT" endlocal
+
+:omega
```
perhaps can be improved?

kailashgautham replied:
Interesting, I'll take a look! Thanks 👍

alfaloo commented on src/main/java/kaiyap/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+                this.deadline + ")"
+                ;
+    }
+}
```
I like the logic

g-tejas replied:
I truly agree

kailashgautham replied:
Thank you! Your words of affirmation mean a lot to me.

alfaloo reviewed (commented):
I like your work, I approve

g-tejas commented on gradle/wrapper/gradle-wrapper.jar:
Are we supposed to push the .jar file?

kailashgautham replied:
Added to the .gitignore, thanks!

g-tejas commented on src/main/java/kaiyap/KaiYap.java:
```diff
@@ -0,0 +1,214 @@
+        } else {
+            taskList.get(numericIndex).setTaskDone(true);
+            System.out.println(
+                    "\t____________________________________________________________\n" +
```
Perhaps can move the seperator line into a static variable so no need to copy paste the long line everywhere?

Like 
```java
static String SEPARATOR = ".....\n";
```

kailashgautham replied:
Great advice, thanks! I've fixed this.

g-tejas commented on src/main/java/kaiyap/Ui.java:
```diff
@@ -0,0 +1,96 @@
+
+public class Ui {
+
+    TaskList taskList;
```
Missing private access modifier

kailashgautham replied:
Fixed, thanks!

g-tejas commented on src/main/java/kaiyap/Parser.java:
```diff
@@ -0,0 +1,26 @@
+
+public class Parser {
+
+    private final ArrayList<String> actions = new ArrayList<>(Arrays.asList("mark", "unmark", "find", "delete"));
```
Perhaps can move to constructor?

kailashgautham replied:
Fixed, thanks!

g-tejas commented on src/main/java/kaiyap/KaiYap.java:
```diff
@@ -0,0 +1,214 @@
+                }
+                break;
+            case "find":
+                try {
```
Instead of try catching every switch case, perhaps can wrap the entire switch statement in a single try catch? (still within the while loop)

kailashgautham replied:
Fixed, thanks! Looks a lot cleaner now 👍

g-tejas reviewed (commented):
LGTM, minor nitpicks

kailashgautham committed:
Add javafx functionality

kailashgautham committed:
JavaFX - Merge branch 'branch-Level-10'

kailashgautham committed:
Add varargs functionality

kailashgautham committed:
Make styling changes

kailashgautham committed:
Add stream functionality

Introduced stream functionality to improve task search efficiency.
This update enhances search performance, using streams.

kailashgautham committed:
Improve code quality by moving print statement to Ui class

Cleaned up code by moving print string to Ui class.
This update cleans up functions and improves
the readability of the code.

kailashgautham committed:
Add assertions in key points of code

Added assertions to document important assumptions in code.

kailashgautham committed:
Merge pull request #1 from kailashgautham/A-Assertions

Add assertions in key points of code

kailashgautham committed:
Update changes - Merge branch 'master' into A-Streams

kailashgautham committed:
Merge code - Merge branch 'master' into A-CodeQuality

kailashgautham committed:
Merge pull request #3 from kailashgautham/A-Streams

Add stream functionality

kailashgautham committed:
Merge code - Merge branch 'master' into A-CodeQuality

kailashgautham committed:
Merge pull request #2 from kailashgautham/A-CodeQuality

Improve code quality by moving print statement to Ui class

kailashgautham committed:
Add checks for existing tasks

kailashgautham committed:
Merge code - Merge branch 'branch-BCD-Extension'

kailashgautham committed:
Add yml file for Github Actions CI

kailashgautham committed:
Modify yml file

kailashgautham committed:
Make checkstyle changes

kailashgautham committed:
Readme changes

kailashgautham committed:
Add features to readme

kailashgautham committed:
Add styling changes

kailashgautham committed:
Fix styling issue in readme

kailashgautham committed:
Add better gui, personality, error handling, testing

kailashgautham committed:
Modify readme

kailashgautham committed:
Fix stylecheck changes

kailashgautham committed:
Fix testing scripts

kailashgautham committed:
Add testing changes

kailashgautham committed:
Testing changes again

kailashgautham committed:
Some more testing changes

kailashgautham committed:
Fix error throwing

kailashgautham committed:
Change boolean naming

kailashgautham committed:
Change styling in Ui.java

kailashgautham committed:
Modify DialogBox comments to be concise

kailashgautham committed:
Improve error handling in KaiYap

kailashgautham committed:
Make error change

kailashgautham committed:
Add ai increment javadoc
